src/StreamMaster.cpp

```diff
@@ -150,6 +152,11 @@ void StreamMaster::closeStream(Streamer &Stream, const std::string &TopicName) {
         TopicName, NumStreamers);
   }
   Stream.close();
+
+  if (NumStreamers == 0) {
```

Comment:
I removed this on a different PR.
I think for testing it is useful to be able to run up a stream master with no streams and not have it close itself.

Reply:
Currently if I remove this code then the writer would never stop and close the file... will discuss with you on Slack.